### ISSUE TITLE
rsub with multiple computers in the same server

### ISSUE THREAD

**Issue by jsmedmar** [henrikpersson/rsub#60]
I'm trying to use rsub with multiple computers on the same server, however, it only works for one. When I try to use it from other computers, files are opened in the first computer. I tried following the instructions in the __Port Forwarding__ section of this [blog](http://blog.macromates.com/2011/mate-and-rmate/). But couldn't solve it. Any suggestions?

**Comment by jsmedmar** [comment]
I found a work around, which is described [here](http://stackoverflow.com/questions/35343853/how-to-use-sublime-text-over-ssh-with-multiple-computers-in-same-server-using-rs/35343854#35343854). I'm wondering why this package does not enable to set use specific settings.

**Comment by lgsds** [comment]
The workaround you linked doesn't seem to be working for me. Any suggestions?

**Comment by jsmedmar** [comment]
@lgsds can you explain further? What are you experiencing? The same file is opening in two computers? 
Nothing happening at all? I'll review my SO post to see if I missed something.

**Comment by jsmedmar** [comment]
Make sure in this file ``Packages/User/rsub.sublime-settings `` you use different ports for the different machines.

**Comment by lgsds** [comment]
When I follow the post I get the following errors:

~/bin/rsub: connect: Connection refused
~/bin/rsub: line 326: /dev/tcp/localhost/52699: Connection refused
Unable to connect to TextMate on localhost:52699

**Comment by jsmedmar** [comment]
Do you have sublime opened when you try to connect? Perhaps, could you share your settings?

S

**Comment by lgsds** [comment]
I do. I set it up using the same settings as the SO post:

# Packages/User/rsub.sublime-settings

// rsub user settings

{
    /*
        rsub listen port.

        IMPORTANT: Use a different port for each machine.
    */
    "port": 52699,

    /*
        rsub listen host

        WARNING: it's NOT recommended to change this option,
        use SSH tunneling instead.
    */
    "host": "localhost"
}

**Comment by jsmedmar** [comment]
Hey @lgsds, I went through this process again for one of my colleagues computer and I was having the same issue as you...

Surprisingly, my problem was that I had an extra white space at the end of the name of the rsub settings file, it was called `Packages/User/rsub.sublime-settings ` instead of `Packages/User/rsub.sublime-settings`.

Once I fixed that, everything worked as expected. I didn't do anything different from what is described in StackOverflow.

**Comment by jsmedmar** [comment]
If you copy pasted  the name from the SO post, you are likely to have the same problem because that name had an extra white space (which I just corrected).

**Comment by inovek** [comment]
In our case there was two local machines which wanted to connect on the same server and the same user. We solved it by simply duplicate the rmate-script on the remote machine, rename it and change the port-number in the script itself.

All other changes locally was also necessary liked mentioned in this thread. Worked like a charm, but a support for multiple users for the same command would be nice.

**Comment by KES777** [comment]
If you connect over ssh you may export `RMATE_PORT` environment variable and do remote forward.
Because I am the only user I put this export into `~/.bash_aliases`
```
if [ -z "$SSH_CLIENT" ]; then
        export EDITOR="subl"
else
        export EDITOR="rsub --port 52697"
        export RMATE_PORT="52697"
fi
```
and also put into `~/.ssh/config` next:
```
host    your_host_alias
        user user_name
        hostname host.domain
        PreferredAuthentications publickey
        port 22
        identityfile /path/to/your/key
        RemoteForward 52697 127.0.0.1:52698
```
How to pass the ENV variable you can see [here](https://superuser.com/a/702751/431840)

for example use `RSUB_FORWARD` environment variable.
you users should setup these variable to `11111` and `22222` correspondingly.
Replace `52697` in `ssh/config` to `11111`/`22222`
and in the `~/.bash_aliases` on remote side:
```
        export EDITOR="rsub --port $RSUB_FORWARD"
        export RMATE_PORT="$RSUB_FORWARD"
```

hope this helps to you

**Comment by magnosis** [comment]
I see how to customize the port, but I fail to understand how to have 1 local Sublime instance able to edit files on multiple remote servers at once.

Every time I open an SSH connection to 2nd, 3rd, etc. remote hosts, the previous hosts cannot connect to Sublime. One local Sublime port won't accept connection from multiple remote hosts.

Is this possible at all? Can I get sublime to open, say, 10 ports, 1 for each of 10 remote hosts?

**Comment by KES777** [comment]
@magnosis 
>One local Sublime port won't accept connection from multiple remote hosts.

You are wrong here. The sublime when listening port 52698 (actually rmate plugin) it acts like server. and can accept unlimited count of connections from different remote hosts. In same way as for example your DB server.

>Is this possible at all? Can I get sublime to open, say, 10 ports, 1 for each of 10 remote hosts?

yes that is possible. I have described [here](https://github.com/henrikpersson/rsub/issues/60#issuecomment-350343053) how
